flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/filesystem/FileSystemTableSink.java

```diff
@@ -158,8 +159,8 @@ public SinkRuntimeProvider getSinkRuntimeProvider(Context sinkContext) {
     private RowDataPartitionComputer partitionComputer() {
         return new RowDataPartitionComputer(
                 defaultPartName,
-                schema.getFieldNames(),
-                schema.getFieldDataTypes(),
+                DataType.getFieldNames(getPhysicalDataType()).toArray(new String[] {}),
```

Comment:
nit: in Flink we usually use `toArray(new String[0])` instead of `toArray(new String[] {})`
see https://stackoverflow.com/questions/9572795/convert-list-to-array-in-java